lib/registrations.js

```diff
@@ -124,15 +134,96 @@ const asClass = (Type, opts) => {
 
   opts = makeOptions(defaults, opts)
 
-  const resolve = (container) => new Type(container.cradle)
-
+  const resolve = generateResolve(buildNewClassFunction(Type), Type.prototype.constructor)
   const result = {
     resolve,
-    lifetime: opts.lifetime
+    lifetime: opts.lifetime,
+    resolutionMode: opts.resolutionMode
   }
-
+  result.resolve = resolve.bind(result)
   Object.assign(result, makeFluidInterface(result))
+
   return result
 }
 
+/**
+ * Returns a function used to invoke a type constructor.
+ *
+ * @param {Class} Type
+ * The type being contructed.
+ *
+ * @return {Function}
+ * The function that constructs the specified type
+ */
+function buildNewClassFunction (Type) {
+  return (dependencies) => {
```

Comment:
I am not sure why I didn't try using `arguments`, it ended up working way better. Thanks!